### PR TITLE
WebClient documentation

### PR DESCRIPTION
**Pull request by Verdent**

### CODE REVIEW

File: docs/se/webclient/introduction.adoc [hunk 1]
```diff
@@ -233,4 +245,77 @@ requestBuilder.readerContext().registerReader(JsonSupport.reader()); // <2>
...
\ No newline at end of file
+<2> Adds JSON-P reader only to this request.
+
+=== WebClient TLS setup
+
+Configure TLS either programmatically or by the Helidon configuration framework.
+
+==== Configuring TLS in your code
+
+The one way to configure TLS in WebClient is in your application code.
```
Comment (ljamen) [inline docs/se/webclient/introduction.adoc:256]: One way to configure TLS in WebClient is in your application code as shown below.

File: docs/se/webclient/introduction.adoc [hunk 2]
```diff
@@ -233,4 +245,77 @@ requestBuilder.readerContext().registerReader(JsonSupport.reader()); // <2>
...
+         .tls(WebClientTls.builder()
+               .certificateTrustStore(keyConfig)
+               .clientKeyStore(keyConfig)
+               .build())
+         .build();
+----
+
+==== Configuring TLS in the config file
+
+It is also possible to configure TLS via the config file.
```
Comment (ljamen) [inline docs/se/webclient/introduction.adoc:279]: Another way to configure TLS in WebClient is through the `application.yaml` configuration file.

**Review by ljamen — COMMENTED** [review]
Small updates, LGTM.

File: docs/includes/config/io.helidon.webclient.Proxy.adoc [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+|`password` |string |{nbsp} |Sets a new password for the proxy.
+|`port` |int |{nbsp} |Sets a port value.
+|`host` |string |{nbsp} |Sets a new host value.
+|`use-system-selector` |boolean |`false` |Configure proxy from environment variables and system properties.
+|`username` |string |{nbsp} |Sets a new username for the proxy.
+|`type` |string (HTTP, SOCKS_4, SOCKS_5, SYSTEM) |`HTTP` |Sets a new proxy type.
+|`no-proxy` |string[&#93; |{nbsp} |Configure a host pattern that is not going through a proxy.
+ <p>
+ Options are:
+ <ul>
```
Comment (tomas-langer) [inline docs/includes/config/io.helidon.webclient.Proxy.adoc:51]: This is not correct. We need to update our generator to create a better table for this (https://github.com/tomas-langer/config-documentation/pull/2).

File: media/common/src/main/java/io/helidon/media/common/MediaContext.java [hunk 4]
```diff
@@ -160,6 +163,11 @@ private Builder() {
          * @param config a {@link Config}
          * @return this {@link Builder}
          */
+        @ConfiguredOption(key = "services",
```
Comment (tomas-langer) [inline media/common/src/main/java/io/helidon/media/common/MediaContext.java:166]: As far as I remember, the services works same as security providers, so it should use the same style:
`@ConfiguredOption(key = "services", kind = ConfiguredOption.Kind.LIST, required = true, provider = true)`

in your case, you need to explicitly define the type as the class of the service provider interface (if it is not on a method that adds it - in this case `MediaSupportProvider`)

File: media/common/src/main/java/io/helidon/media/common/MediaContext.java [hunk 5]
```diff
@@ -252,6 +262,7 @@ public Builder discoverServices(boolean discoverServices) {
          * @param filterServices filter services
          * @return this builder instance
          */
+        @ConfiguredOption(key = "filter-services")
```
Comment (tomas-langer) [inline media/common/src/main/java/io/helidon/media/common/MediaContext.java:265]: please remove the key, it is automatically created from method name

File: media/common/src/main/java/io/helidon/media/common/MediaContext.java [hunk 6]
```diff
@@ -240,6 +249,7 @@ public Builder registerDefaults(boolean registerDefaults) {
          * @param discoverServices use Java Service Loader
          * @return this builder instance
          */
+        @ConfiguredOption(key = "discover-services")
```
Comment (tomas-langer) [inline media/common/src/main/java/io/helidon/media/common/MediaContext.java:252]: please remove the key, it is automatically created from method name

File: media/common/src/main/java/io/helidon/media/common/MediaContext.java [hunk 7]
```diff
@@ -229,6 +237,7 @@ public Builder addStreamWriter(MessageBodyStreamWriter<?> streamWriter) {
          * @param registerDefaults register defaults
          * @return this builder instance
          */
+        @ConfiguredOption(key = "register-defaults", description = "Whether default readers and writers should be registered")
```
[the remark this answers is not in the document]
Comment (tomas-langer) [reply media/common/src/main/java/io/helidon/media/common/MediaContext.java:240]: Why did you not modify the javadoc of the method? It seems the extended description would be benefitial for both (and then you can remove it from the annotation)

File: webclient/webclient/src/main/java/io/helidon/webclient/Proxy.java [hunk 8]
```diff
@@ -498,6 +502,14 @@ public Builder config(Config config) {
          * @param type proxy type
          * @return updated builder instance
          */
+        @ConfiguredOption(type = String.class,
```
[the remark this answers is not in the document]
Comment (tomas-langer) [reply webclient/webclient/src/main/java/io/helidon/webclient/Proxy.java:505]: Enums are supported and automatically processed for a list of values

File: webclient/webclient/src/main/java/io/helidon/webclient/Proxy.java [hunk 9]
```diff
@@ -574,6 +591,7 @@ public Builder addNoProxy(String noProxyHost) {
          * @param useIt use system selector
          * @return updated builder instance
          */
+        @ConfiguredOption(key = "use-system-selector", value = "false")
```
Comment (tomas-langer) [inline webclient/webclient/src/main/java/io/helidon/webclient/Proxy.java:594]: please remove the key, it is automatically created from method name

File: webclient/webclient/src/main/java/io/helidon/webclient/WebClientConfiguration.java [hunk 10]
```diff
@@ -359,6 +364,7 @@ public B readTimeout(Duration readTimeout) {
          * @param followRedirects follow redirection
          * @return updated builder instance
          */
+        @ConfiguredOption(key = "follow-redirects", description = "Whether redirects should be followed or not")
```
[the remark this answers is not in the document]
Comment (tomas-langer) [reply webclient/webclient/src/main/java/io/helidon/webclient/WebClientConfiguration.java:367]: Why create a new description when javadoc seems to be oK?

**Review by tomas-langer — DISMISSED** [review]

**Review by m0mus — DISMISSED** [review]

**Review by m0mus — APPROVED** [review]

**Pull request merged**